### PR TITLE
Set arguments types dynamically from JSON

### PR DESCRIPTION
**Pull request by DieMyst**
It fixes types resolving in `aqua run` for a case where argument uses for `via` route

### CODE REVIEW

File: cli/.js/src/main/scala/aqua/ArgOpts.scala [hunk 1]
```diff
@@ -73,21 +73,44 @@ object ArgOpts {
...
+      case a if js.typeOf(a) == "boolean" => validNec(ScalarType.bool)
+      case a if js.Array.isArray(a) =>
+        // if all types are similar it will be array array with this type
+        // otherwise array with bottom type
+        val elementsTypesV: ValidatedNec[String, List[Type]] =
+          a.asInstanceOf[js.Array[js.Dynamic]].map(ar => jsTypeToAqua(name, ar)).toList.sequence
+
+        elementsTypesV.andThen { elementsTypes =>
+          if (elementsTypes.isEmpty) validNec(ArrayType(BottomType))
+          else if (elementsTypes.forall(_ == elementsTypes.head))
```
Comment (alari) [inline cli/.js/src/main/scala/aqua/ArgOpts.scala:98]: What will happen if the array is empty?

File: cli/.js/src/main/scala/aqua/ArgOpts.scala [hunk 2]
```diff
@@ -96,18 +119,31 @@ object ArgOpts {
...
+
+          typeV.map(t => (vm.copy(baseType = t), arg))
+        }.sequence
+          .map(_.map { case (vm, arg) =>
+            vm.name -> ArgumentGetter(vm, arg)
+          }.toMap)
+          .andThen { services =>
+            val argsWithTypes = cliFunc.args.map {
+              case v @ VarRaw(n, t) =>
+                services.get(n).map(g => v.copy(baseType = g.function.value.baseType)).getOrElse(v)
```
Comment (alari) [inline cli/.js/src/main/scala/aqua/ArgOpts.scala:141]: I cannot understand this line

**Review by alari — APPROVED** [review]

File: cli/.js/src/main/scala/aqua/ArgOpts.scala (hunk 2 above)
Comment (DieMyst) [reply cli/.js/src/main/scala/aqua/ArgOpts.scala:141]: If there is a service with such name, get type from this service and put it to argument

Comment (alari) [reply cli/.js/src/main/scala/aqua/ArgOpts.scala:141]: Why? Can you add a comment to this line?

File: cli/.js/src/main/scala/aqua/ArgOpts.scala (hunk 1 above)
Comment (DieMyst) [reply cli/.js/src/main/scala/aqua/ArgOpts.scala:98]: everything will be ok, cause we already know that it is an array

**Pull request merged**